utils/common/src/main/kotlin/ArchiveUtils.kt

```diff
@@ -87,13 +87,13 @@ fun archive(inputDir: File, zipFile: File, prefix: String = ""): File =
 /**
  * Unpack the [File] to [targetDirectory] using [filter] to select only the entries of interest. If an archive type
  * other than [ArchiveType.NONE] is specified in [forceArchiveType], use this one; otherwise, try to detect the type
- * based on the file extension.
+ * based on the file extension. Return a flag whether the archive was not empty and could be processed.
```

Comment:
Hmm. I guess at least for some archive types an empty archive is still valid archive, so I'd expect `true` to be returned in that case.

Reply:
Also, what if the archive is corrupt? According to the docs I'd assume `false` to be returned in that case, whereas instead probably an exception is thrown by the downstream function.

So, what about simply determining the return value by wrapping the `when` in `runCatching` and returning `true` on success, and `false` on failure / if an exception was thrown? That would avoid the changes to the type-specific `unpack` functions.

Reply:
The problem I want to solve with this approach is that for some archive types no exception is thrown, even if the archive has a different format. So, the only indication whether the archive type is correct in such cases is that at least one entry can be extracted.

Would you prefer throwing an exception in these cases instead of returning a flag?

Reply:
> Would you prefer throwing an exception in these cases instead of returning a flag?

As discussed separately, yes.

Reply:
> further tests have shown that only archive formats operating on `ArchiveInputStream` are affected

Thanks for finding this out!